### PR TITLE
Move settings updates into a queue.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Followup to https://github.com/dotnet/roslyn/pull/74247

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@ToddGrun ptal.

File: src/EditorFeatures/Core/EditorConfigSettings/Aggregator/SettingsAggregatorFactory.cs [hunk 1]
```diff
@@ -4,20 +4,29 @@
...
-internal class SettingsAggregatorFactory : IWorkspaceServiceFactory
+internal sealed class SettingsAggregatorFactory : IWorkspaceServiceFactory
 {
+    private readonly IThreadingContext _threadingContext;
+    private readonly IAsynchronousOperationListener _listener;
+
     [ImportingConstructor]
     [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-    public SettingsAggregatorFactory()
+    public SettingsAggregatorFactory(
```
Comment (ToddGrun) [inline src/EditorFeatures/Core/EditorConfigSettings/Aggregator/SettingsAggregatorFactory.cs:22]: >public SettingsAggregatorFactory( [](http://example.com/codeflow?start=4&length=33)

nit: isn't your reflex to switch these to default ctor syntax?

File: src/EditorFeatures/Core/EditorConfigSettings/Aggregator/SettingsAggregator.cs [hunk 2]
```diff
@@ -3,31 +3,49 @@
...
+        IAsynchronousOperationListener listener)
     {
         _workspace = workspace;
         _workspace.WorkspaceChanged += UpdateProviders;
-        _whitespaceProvider = GetOptionsProviderFactory<Setting>(_workspace);
-        _codeStyleProvider = GetOptionsProviderFactory<CodeStyleSetting>(_workspace);
-        _namingStyleProvider = GetOptionsProviderFactory<NamingStyleSetting>(_workspace);
-        _analyzerProvider = GetOptionsProviderFactory<AnalyzerSetting>(_workspace);
+
+        //
```
Comment (ToddGrun) [inline src/EditorFeatures/Core/EditorConfigSettings/Aggregator/SettingsAggregator.cs:36]: >//  [](http://example.com/codeflow?start=8&length=3)

nit: comment MIA :)

**Review by ToddGrun — APPROVED** [review]
:shipit:

File: src/EditorFeatures/Core/EditorConfigSettings/Aggregator/SettingsAggregatorFactory.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core/EditorConfigSettings/Aggregator/SettingsAggregatorFactory.cs:22]: yes.  will do in that other mop up pr.

File: src/EditorFeatures/Core/EditorConfigSettings/Aggregator/SettingsAggregator.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core/EditorConfigSettings/Aggregator/SettingsAggregator.cs:36]: yes. will do in that other mop up pr.

**Pull request merged**